ui/pages/swaps/view-quote/view-quote.js

```diff
@@ -214,10 +263,21 @@ export default function ViewQuote() {
     );
   }
 
-  const gasTotalInWeiHex = calcGasTotal(
-    maxGasLimit,
-    networkAndAccountSupports1559 ? maxFeePerGas : gasPrice,
-  );
+  // Smart Transactions gas fees.
+  if (
+    smartTransactionsEnabled &&
+    smartTransactionsOptInStatus &&
+    unsignedTransactionsAndEstimates
+  ) {
+    const maxFeePerGasDecWEI =
+      unsignedTransactionsAndEstimates.fees[
+        unsignedTransactionsAndEstimates.fees.length - 1
+      ]?.maxFeePerGas;
+    maxFeePerGas = decWEIToHexWEI(maxFeePerGasDecWEI);
+    maxGasLimit = unsignedTransactionsAndEstimates.gasLimit;
+  }
+
+  const gasTotalInWeiHex = calcGasTotal(maxGasLimit, maxFeePerGas || gasPrice);
```

Comment:
I need to make sure that all fee estimates for regular / smart transactions show correct values. We especially need to complete this pending item: Check why we see a big gap between estimated and max fee on Rinkeby